zerver/lib/narrow.py

```diff
@@ -676,6 +679,25 @@ def by_dm_including(
         )
         return query.where(maybe_negate(cond))
 
+    def by_mention(
+        self, query: Select, operand: str | int, maybe_negate: ConditionTransform
+    ) -> Select:
+        assert self.user_profile is not None
+
+        try:
+            if isinstance(operand, int):
+                user = get_user_by_id_in_realm_including_cross_realm(int(operand), self.realm)
+            else:
+                user = access_user_by_email(self.user_profile, operand, for_admin=False)
```

Comment:
As discussed above, let's just do user IDs in the server. The old email format is only for backwards-compatibility, which is not required here as this is a new feature.